clang/lib/3C/TypeVariableAnalysis.cpp

```diff
@@ -96,6 +97,9 @@ bool TypeVarVisitor::VisitCallExpr(CallExpr *CE) {
     if (FDef == nullptr)
       FDef = FD;
     if (auto *FVCon = Info.getFuncConstraint(FDef, Context)) {
+      // if we need to rewrite it but can't (macro, etc), it isn't safe
+      bool ForcedInconsistent = !typeArgsProvided(CE)
+                                && !canRewrite(*CE,*Context);
```

Comment:
`ProgramInfo::constrainWildIfMacro` uses `Rewriter::isRewritable` for a similiar check, so I think it would be more consistent to use it here as well. I don't know how exactly this differs from `canRewrite`, but I think for now I'd like to keep the use of `canRewrite` in the rewriter logic as long as making this change doesn't break anyhting.

Reply:
Done. I guess I found the other one first. There no difference in the regression tests.